infor-design/enterprise
# 4987/4926/4964 - Fixed date, time and locale issues

**Explain the _details_ for making this change. What existing problem does the pull request solve?**
Fixed time and locale issues.
- Time was changing, if selected time was before noon for Danish language locale `da-DK`
- Time format `HHmm` was not working for time picker editor with Datagrid
- Date and available date validation was not working for Croatian locale `hr-HR` with Date picker

**Related github/jira issue (required)**:
Closes #4987
Closes #4926
Closes #4964

**Steps necessary to review your pull request (required)**:
Pull this branch and build/run the demo app

Test: 4987
- Navigate to: http://localhost:4000/components/datepicker/example-timeformat.html?locale=da-DK
- Click on 3rd from top picker labeled as `Date Time Picker in Locale (Time to Current)` to open
- Select and set time as `11.25`
- Click on button on `Anvend` to close picker
- Now click anywhere on the page to get off the focus
- See the time should NOT changes

Test: 4926
- Navigate to: http://localhost:4000/components/datagrid/test-editable-time
- Click on trigger icon in any cell in column `Time 24 (no separator)` to open picker
- In picker should have correct value
- Change and pick some time, after close picker it should set the correct time on filed (without separator HHmm formate)

Test: 4964
- Navigate to: http://localhost:4000/components/datepicker/example-timeformat.html?locale=hr-HR
- Use any picker on page and set some date
- It should not show any validation error

**Included in this Pull Request**:
